File: sdks/php/src/Invoices/Requests/SendEmailInvoicesRequest.php
<?php

namespace NovaBilling\Invoices\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;

class SendEmailInvoicesRequest extends JsonSerializableType
{
    /**
     * @var ?string $email Recipient email address. Defaults to the customer email if omitted.
     */
    #[JsonProperty('email')]
    public ?string $email;

    /**
     * @param array{
     *   email?: ?string,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->email = $values['email'] ?? null;
    }
}

File: sdks/csharp/src/NovaBilling/Invoices/Requests/SendEmailInvoicesRequest.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record SendEmailInvoicesRequest
{
    /// <summary>
    /// Invoice ID
    /// </summary>
    [JsonIgnore]
    public required string Id { get; set; }

    /// <summary>
    /// Recipient email address. Defaults to the customer email if omitted.
    /// </summary>
    [JsonPropertyName("email")]
    public string? Email { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
